### DISCUSSION TITLE
comunica-browser fails when built locally

### DISCUSSION THREAD

**Discussion by larsgsvensson** [comunica/comunica#150] [Q&A] [answered]
#### Issue type:

- :bug: Bug

____
#### Description:

Having managed to build `comunica-browser.js` correctly (thanks!), I failed at building a small web application querying the DBPedia endpoint. HTML source code:

```html
<html>
	<head>
		<title>Test SPARQL Client</title>
		<script src="js/comunica-browser.js"></script>
		<script>
		function search() {
		Comunica.evaluateQuery(document.getElementById('query').value,
			{ sources: [ { type: 'hypermedia', value: 'http://fragments.dbpedia.org/2015/en' } ] })
		.then(function (result) {
			result.bindingsStream.on('data', function (data) {
				console.log(data.toObject());
				var resultsTable = document.getElementById('resultsTable');
				var row = resultsTable.insertRow(-1);
				var cell = row.insertCell(0);
				cell.innerHTML = data.toObject();
			});
		});}
		</script>
	</head>
	<body>
		<div id="searchbox">
			<form id="searchForm" onsubmit="search()" accept-charset="UTF-8">
				<textarea id="query" >SELECT * { ?s ?p &lt;http://dbpedia.org/resource/Belgium&gt;. ?s ?p ?o } LIMIT 100</textarea>
				<input type="submit" value="Search"></input>
			</form>
		</div>
		<div id="results">
			<table id="resultsTable">
			</table>
		</div>
	</body>
</html>
```
However, when I hit "Search" the console does not display any search results but reference and syntax errors in `comunica-browser-js`, `index.js`, `context.js`, `es6-promise.js`, `Requester-browser.js`, `compact.js` and `json-ld.js`.
____
#### Environment:

Firefox Quantum 60.0.2 (64-bit)

#### Log file:

https://pastebin.com/cyxJP6AC

**Comment by rubensworks** [comment]
Did you by any chance get any errors or warnings when running the `yarn run browser` command?

What happens when you query using http://comunica.linkeddatafragments.org/? Does that work correctly for you?

**Comment by larsgsvensson** [comment]
> Did you by any chance get any errors or warnings when running the yarn run browser command?

Yes, but none that seemed to have anything to do with the functionality. Command output at https://pastebin.com/vE0FuJQS

> What happens when you query using http://comunica.linkeddatafragments.org/? Does that work correctly for you?

The query
```sparql
SELECT * { ?s ?p &lt;http://dbpedia.org/resource/Belgium&gt;. ?s ?p ?o } LIMIT 100
```
returns results that seem correct.

**Comment by rubensworks** [comment]
The webpack output indeed looks fine.

Could you send your generated `comunica-browser.js` file? I'll have a look at it then.

**Comment by larsgsvensson** [comment]
Thanks, it's uploaded as [comunica-browser.zip](https://github.com/comunica/comunica/files/2089711/comunica-browser.zip) and includes the generated `comunica-browser.js.map`

**Comment by rubensworks** [comment, marked as answer]
I just tested your uploaded browser script with the HTML document. The JS works fine for me, but there were some errors in the HTML file.

Making the following changes in the HTML results in the expected behaviour:
* At the end of your `search()` function definition, make sure to return `false`, so that the default form behaviour is blocked, which would result in a page refresh.
* Change your `onsubmit` to `onsubmit="return search()"`, so that your return value actually has effect.

Be sure to let me know if you would encounter any other problems with the browser script.

**Comment by larsgsvensson** [comment]
Oh, that's embarassing... Thanks @rubensworks, I'll make sure to update my JS skills... The rest runs nicely (but for #121 that makes the client show the metadata triples, too... but that's another story)
